src/main/java/org/sunbird/workflow/service/impl/WorkFlowServiceImplV2.java

```diff
@@ -523,4 +518,54 @@ private void addModificationEntry(WfStatusEntity applicationStatus, String userI
         }
 
     }
+
+    public void processAndUpdateProfessionalDetails(String userId, String updateFieldValuesJson) {
+        try {
+            JsonNode updateFieldValuesArray = mapper.readTree(updateFieldValuesJson);
+
+            for (JsonNode updateEntry : updateFieldValuesArray) {
+                String fieldKey = updateEntry.get(Constants.FIELD_KEY).asText();
+
+                if (Constants.PROFESSIONAL_DETAILS.equals(fieldKey)) {
+                    JsonNode toValueNode = updateEntry.get(Constants.TO_VALUE);
+
+                    String group = toValueNode.has(Constants.GROUP) ? toValueNode.get(Constants.GROUP).asText() : null;
+                    String designation = toValueNode.has(Constants.DESIGNATION) ? toValueNode.get(Constants.DESIGNATION).asText() : null;
+
+                    updateProfessionalDetailsInCache(userId, group, designation);
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error processing update_field_values JSON", e);
+        }
+    }
+
+    public void updateProfessionalDetailsInCache(String userId, String group, String designation) {
```

Comment:
No need to read from cache... just push the latest data after updating into user profile...